test/Swap.test.ts

```diff
@@ -29,21 +33,47 @@ describe("Swap RBTC", function () {
     const senderBalance: BigNumber = await sideTokenBtc.balanceOf(sender.address);
 
     expect(senderBalance.toString()).to.equals(oneEther.toString());
-    await sideTokenBtc.connect(sender).approve(swapRBTC.address, halfEher);
+    await sideTokenBtc.connect(sender).approve(swapRBTC.address, halfEther);
 
     await deployer.sendTransaction({
       to: swapRBTC.address,
       value: oneEther
     });
 
     const balanceBeforeSwap = await sender.getBalance();
-    const receipt = await swapRBTC.connect(sender).swapWRBTCtoRBTC(halfEher, sideTokenBtc.address);
+    const receipt = await swapRBTC.connect(sender).swapWRBTCtoRBTC(halfEther, sideTokenBtc.address);
     const balanceAfterSwap = await sender.getBalance();
 
     const tx = await network.provider.send("eth_getTransactionReceipt", [receipt.hash]);
     const effectiveGasPrice = BigNumber.from(tx.effectiveGasPrice);
     const gasUsed = BigNumber.from(tx.gasUsed);
 
-    expect(balanceAfterSwap.add(effectiveGasPrice.mul(gasUsed)).sub(halfEher.add(balanceBeforeSwap)).isZero()).to.be.true;
+    expect(balanceAfterSwap.add(effectiveGasPrice.mul(gasUsed)).sub(halfEther.add(balanceBeforeSwap)).isZero()).to.be.true;
+  });
+
+  it("Should Revert swap when transfer amount exceeds allowance", async function () {
+    await sideTokenBtc.connect(minter).mint(sender.address, halfEther, "0x", "0x");
+    const senderBalance: BigNumber = await sideTokenBtc.balanceOf(sender.address);
+
+    expect(senderBalance.toString()).to.equals(halfEther.toString());
+    await sideTokenBtc.connect(sender).approve(swapRBTC.address, quarterEther);
+
+    await deployer.sendTransaction({
+      to: swapRBTC.address,
+      value: oneEther
+    });
+
+    await expect(swapRBTC.connect(sender).swapWRBTCtoRBTC(halfEther, sideTokenBtc.address))
+      .to.be.revertedWith("ERC777: transfer amount exceeds allowance");
+  });
+
+  it("Should Not be allowed to withdraw RBTC When balance is not enough", async function () {
+    await sideTokenBtc.connect(minter).mint(sender.address, halfEther, "0x", "0x");
```

Comment:
I think it's not necesary to mint tokens to the sender for this test

Reply:
You're right. Fixed.